### PR TITLE
Skip TestClusterDNS on GKE (since DNS is disabled there)

### PR DESCRIPTION
**Pull request by zmerlynn**
We'll eventually want to do something fancier here. But the yak needs
to stop growing hair.

### CODE REVIEW

File: cmd/e2e/e2e.go [hunk 1]
```diff
@@ -447,6 +447,11 @@ func outputTAPSummary(infoList []TestInfo) {
 
 // TestClusterDNS checks that cluster DNS works.
 func TestClusterDNS(c *client.Client) bool {
+	if os.Getenv("KUBERNETES_PROVIDER") == "gke" {
```
Comment (thockin) [inline cmd/e2e/e2e.go]: TODO: remove this ASAP

maybe file a followup issue to do this?

**Comment by zmerlynn** [comment]
PTAL.

**Comment by zmerlynn** [comment]
(Travis is a flake)

**Comment by thockin** [comment]
LGTM

**Comment by thockin** [comment]
FYI docs/devel/collab.md - this was a breakage, so OK

**Comment by zmerlynn** [comment]
If the complaint was hours, I'm sitting here watching Jenkins. :) (But
thanks for the heads up.)

**Comment by thockin** [comment]
Because so many of us are active on work hours, we try to make sure
anything that peopl emight want to review does not go in before they have a
reasonable chance.

**Pull request merged**